text/2094-nll.md

```diff
@@ -1674,40 +1764,49 @@ fn access_legal(lvalue, is_shallow, is_read) {
+2. the loan path is, or has a **shallow prefix** `<base>.<field1>` that accesses a field of a union, and `lvalue` has a prefix of the form `<base>.<field2>` for a *different* field with the same base union.
+  - so: if `a.b` is a union with distinct fields `x` and `y`, shallowly accessing a path `a.b.x.c` is illegal if `a.b.y` or `a.b.y.s` is borrowed. There `<base>` is `a.b`, `<field1>` is `y` and `<field2>` is `x`.
+  - but: unless `a.b.x` is *also* a union, it is legal to access the `a.b.x.c` is *legal* if `a.b.x.d` is borrowed, because the same union field is used in both borrows.
+  - the prefix of `lvalue` can be an arbitrary prefix - if `a.b.y.s` is borrowed, then it is illegal to shallowly access `*(*a.b.x.c).t`.
+3. `lvalue` is a **shallow prefix** of the loan path
   - shallow prefixes are found by stripping away fields, but stop at
     any dereference
-  - so: writing a path like `a` is illegal if `a.b` is borrowed
+  - so: writing to a path like `a` is illegal if `a.b` is borrowed
   - but: writing `a` is legal if `*a` is borrowed, whether or not `a`
     is a shared or mutable reference
 
+In addition, if `lvalue` has a prefix that is a dereference `*base_lv`, then the base pointer `base_lv` must be definitely initialized (inductively, checking the shallow prefix should be equivalent)
+ - so: writing to a path like `(*a).x` is legal only if `a` is initialized, and still legal if `(*a).x` itself is not initialized.
+
 For **deep** accesses to the path `lvalue`, we consider borrows relevant
 if they meet one of the following criteria:
 
-- there is a loan for the path `lvalue`;
-  - so: reading a path like `a.b.c` is illegal if `a.b.c` is mutably borrowed
-- there is a loan for some prefix of the path `lvalue`;
-  - so: reading a path like `a.b.c` is illegal if `a` or `a.b` is mutably borrowed
-- `lvalue` is a **supporting prefix** of the loan path
+1. the loan path is a prefix of the path `lvalue`;
+  - so: reading a path like `a.b.c` is illegal if `a`, `a.b` or `a.b.c` is mutably borrowed
+2. the loan path is, or has a **supporting prefix** `<base>.<field1>` that accesses a field of a union, and `lvalue` has a prefix of the form `<base>.<field2>` for a *different* field with the same base union.
+  - so: if `a.b` is a union with distinct fields `x` and `y`, deeply accessing a path `a.b.x.c` is illegal if `a.b.y`, `a.b.y.s`, or (in contrast with shallow accesses, and as long as both dereferences are of a `&mut T`) `*(*a.b.y.s).t` is borrowed. There `<base>` is `a.b`, `<field1>` is `y` and `<field2>` is `x`.
+3. `lvalue` is a **supporting prefix** of the loan path
   - supporting prefixes were defined earlier
   - so: reading a path like `a` is illegal if `a.b` is mutably
     borrowed, but -- in contrast with shallow accesses -- reading `a` is also
     illegal if `*a` is mutably borrowed
-    
+
+And the accessed data and all children in its lvalue tree must be definitely initialized.
```

Comment:
Is this sentence fragment supposed to be connected to the "3." bullet point immediately above, or something higher up?